src/utils/hooks.ts

```diff
@@ -12,3 +12,19 @@ export const useIsMobile = () =>
   useMemo(() => {
     return isMobile()
   }, [])
+
+/**
+ * Returns { has: true, inc: incrementCount: () => void } when incrementCount
+ * has been invoked `targetCount` times
+ * @param targetCount target count
+ */
+export const useHasCounted = (targetCount: number) => {
```

Comment:
Did you end up using this - I image you wanted to count then limit the number of request for profile image 404?